recipes-bsp/u-boot/u-boot-distro-boot.bbappend

```diff
@@ -129,19 +133,23 @@ UENV_VARS_TO_DEL_EXTRA ?= "kernel_image \
     temp \
 "
 
-do_compile:append () {
-    # Normalize input to a shell variable, otherwise following line causes yocto parser to error
-    bank_word_in="${BANK_WORD}"
-    bank_word=$(IFS=';'; i=1; for bw in ${bank_word_in}; do echo "fuse_bank_word_${i}=${bw}"; let "i=i+1"; done)
-    # Replace implicit newlines with explicit newlines
-    bank_word=$(echo "$bank_word" | sed -n -e '1h;2,$H;${g;s/\n/\\n/gp}')
+python get_fuse_bank_word() {
+    bank_word_in=list(eval(d.getVar("BANK_WORD")))
+    bank_word=""
+    for i, bw in enumerate(bank_word_in, start=1):
+        bank_word+="fuse_bank_word_{}={} {};\\n".format(i, bw[0], bw[1])
+    d.setVar("FUSE_BANK_WORDS", bank_word)
+}
+do_compile[prefuncs] += "get_fuse_bank_word"
+get_fuse_bank_word[doc] = "Process BANK_WORD variable to embedd into u-boot script."
```

Comment:
Some suggestions:

- Avoid changing the general format of the input.
- Avoid using a prefunc; at least I don't see the need for it if we simply set another variable from `BANK_WORD`.
- Since the value in `FUSE_BANK_WORD` is very specific (it's the string to the added to uEnv.txt already formatted to be used by `sed`), I'd use a name indicating it's usage is somehow internal e.g. `BANK_WORD_INTERNAL`.

Something like this:

```
def get_bank_word_internal(d):
    bank_word_in = d.getVar("BANK_WORD").split(";")
    bank_word = ""
    for i, bw in enumerate(bank_word_in, start=1):
        bank_word += "fuse_bank_word_{}={};\\n".format(i, bw)
    return bank_word

BANK_WORD_INTERNAL = "${@get_bank_word_internal(d)}"

# then use BANK_WORD_INTERNAL in do_compile()
```

We'd need to remove the last `;` from `BANK_WORD` with this implementation though.

What do you think?

Reply:
I think itens 2 and 3 do make sense, and I'll submit the changes.
About 1, I changed the format of the variables for 2 reasons:
 1. Since BANK_WORD is a string, whenever we parse we would get all the words in the string. For instance, with the split on ';' we would get: '3', ' ', '0', '3', ' ', '1'...
 2. Since this is an arbitrary variable that we create and consume in the same file.
So because of this, I decided to already initialize those as a tuple, since it would facilitate the processing.

Reply:
I'm not sure I understand what you mean here. The result of `"3 0;3 1;...;3 7".split(";")` is `['3 0', '3 1', ..., '3 7']` which seems to be exactly what we need, no?

Reply:
With `.split(";")`, it was adding a last empty line.
```
WARNING: u-boot-distro-boot-1.0-r0 do_compile: internal fuse_bank_word_1=3 0;
fuse_bank_word_2= 3 1;                                                                                   
fuse_bank_word_3= 3 2;                                                                                   
fuse_bank_word_4= 3 3;                                                                                   
fuse_bank_word_5= 3 4;                                                                                   
fuse_bank_word_6= 3 5;                                                                                   
fuse_bank_word_7= 3 6;                                                                                   
fuse_bank_word_8= 3 7;
fuse_bank_word_9=;
```

So, since how we setup these variables are not that important, I'll keep the tuple format. Just so we can fix this issue and go back to having our nightlies.

Reply:
np, approving it